aten/src/ATen/cpu/vec/sve/vec_common_sve.h

```diff
@@ -63,15 +94,15 @@ DEFINE_FLOAT_INT_CAST(int16_t, 16, float, 32)
 template<int64_t scale = 1>
 std::enable_if_t<scale == 1 || scale == 2 || scale == 4 || scale == 8, Vectorized<double>>
 inline gather(const double* base_addr, const Vectorized<int64_t>& vindex_) {
-  svint64_t vindex = svasrd_n_s64_x(ptrue, svmul_s64_x(ptrue, vindex_, svdup_n_s64(scale)), 3);
```

Comment:
not obvious to me what's motivating the changes in this file; ideally another reviewer who actually knows SVE needs to take a look

Reply:
Hi @swolchok 
Thank you for pointing this out. Here's breakdown of motivation behind why i make changes in this file:


_Change in Gather and mask gather_ 

**Removal of Manual Shifting:**
- Initial: Used svasrd_n_s64_x with shifts by 3 (for double) and 2 (for float) to manually adjust scaled indices.
- Modified: Removed the shifting entirely, using svmul_* to directly compute offsets.

```
// Initial
svint64_t vindex = svasrd_n_s64_x(ptrue, svmul_s64_x(ptrue, vindex_, svdup_n_s64(scale)), 3);

// Modified
svint64_t offsets = svmul_s64_x(ptrue, vindex_, svdup_n_s64(scale));
```

- Reason: The shifts were redundant since svmul_* already computes the correctly scaled 
offsets required by gather operations, eliminating unnecessary arithmetic operations.

**Switch to Offset-Based Gathering:**
- Initial: Used svld1_gather_*index, which expects raw indices.
- Modified: Replaced with svld1_gather_*offset, which uses pre-computed byte offsets.
```
// Initial
return svld1_gather_s64index_f64(ptrue, base_addr, vindex);

// Modified 
return svld1_gather_s64offset_f64(ptrue, base_addr, offsets);
```
- Reason: Offset-based gathering simplifies addressing logic, avoids additional scaling during execution, and aligns better with modern SIMD practices.

**Improved Mask Handling:**
- Initial: Combined mask handling and gathering in a single line and used a generic mask variable.
- Modified: Separated gathering and selection operations into distinct steps and renamed mask to valid_mask for clarity.
```
// Initial
return svsel_f64(mask, svld1_gather_s64index_f64(mask, base_addr, vindex), src);

// Modified
svfloat64_t gathered = svld1_gather_s64offset_f64(valid_mask, base_addr, offsets);
return svsel_f64(valid_mask, gathered, src);
```
- Reason: This separation improves code readability, makes the logic easier to debug, and ensures correctness by explicitly handling each step.